File: frontend/src/types/analysis.ts
// TODO: Replace unknown[] and Record<string, unknown> with Plotly.Data[] and
// Partial<Plotly.Layout> once react-plotly.js and @types/plotly.js are installed in Plan 04.

export interface CoefficientRow {
  variable: string;
  estimate: number;
  std_error: number;
  t_stat: number;
  p_value: number;
  ci_lower: number;
  ci_upper: number;
}

export interface ModelSummary {
  r_squared: number;
  adj_r_squared: number;
  f_statistic: number;
  f_p_value: number;
  n_obs: number;
  degrees_of_freedom: number;
}

export interface DiagnosticResult {
  statistic: number;
  p_value: number | null;
  df?: number | null;
}

export interface DiagnosticsBundle {
  breusch_pagan: DiagnosticResult;
  durbin_watson: DiagnosticResult;
  vif: Record<string, number>;
  shapiro_wilk: DiagnosticResult;
}

export interface ChartData {
  name: string;
  data: unknown[];
  layout: Record<string, unknown>;
}

export interface FollowUpSuggestion {
  title: string;
  explanation: string;
  prompt_text: string;
}

export interface AnalysisResult {
  job_id: string;
  status: "success";
  coefficients: CoefficientRow[];
  model_summary: ModelSummary;
  diagnostics: DiagnosticsBundle;
  charts: ChartData[];
  r_code: string;
  interpretation: string;
  follow_up_suggestions: FollowUpSuggestion[];
}

export interface AnalysisError {
  job_id: string;
  status: "error";
  error_explanation: string;
  suggested_prompt: string;
  r_stderr: string;
  r_code: string;
}

export type AnalysisResponse = AnalysisResult | AnalysisError;

File: backend/app/schemas/analysis.py
from pydantic import BaseModel, ConfigDict


class AnalysisRunRequest(BaseModel):
    model_config = ConfigDict(strict=True)
    job_id: str          # UUID of the preview_ready job
    prompt: str          # Original user prompt text


class CoefficientRow(BaseModel):
    variable: str
    estimate: float
    std_error: float
    t_stat: float
    p_value: float
    ci_lower: float
    ci_upper: float


class ModelSummary(BaseModel):
    r_squared: float
    adj_r_squared: float
    f_statistic: float
    f_p_value: float
    n_obs: int
    degrees_of_freedom: int


class DiagnosticResult(BaseModel):
    statistic: float
    p_value: float | None = None   # DW p-value can be null
    df: int | None = None          # Only BP has df


class DiagnosticsBundle(BaseModel):
    breusch_pagan: DiagnosticResult
    durbin_watson: DiagnosticResult
    vif: dict[str, float]          # variable_name -> VIF value; empty dict if single predictor
    shapiro_wilk: DiagnosticResult


class ChartData(BaseModel):
    name: str              # "coefficient_plot" or "residual_plot"
    data: list[dict]       # Plotly data array (parsed from R plotly_json)
    layout: dict           # Plotly layout object (parsed from R plotly_json)


class FollowUpSuggestion(BaseModel):
    title: str
    explanation: str
    prompt_text: str


class AnalysisResultResponse(BaseModel):
    """Full successful analysis result returned to frontend."""
    model_config = ConfigDict(from_attributes=True)
    job_id: str
    status: str                              # "success"
    coefficients: list[CoefficientRow]
    model_summary: ModelSummary
    diagnostics: DiagnosticsBundle
    charts: list[ChartData]
    r_code: str                               # The generated R script
    interpretation: str                       # Claude Stage 2 plain-English text
    follow_up_suggestions: list[FollowUpSuggestion]


class AnalysisErrorResponse(BaseModel):
    """Error result when R fails."""
    model_config = ConfigDict(from_attributes=True)
    job_id: str
    status: str                              # "error"
    error_explanation: str                   # Claude plain-English error explanation
    suggested_prompt: str                    # Claude's suggested fix
    r_stderr: str                            # Raw R error output
    r_code: str                              # The R script that failed
